### ISSUE TITLE
buenos_aires crashes because ZIP file is named unexpectedly 

### ISSUE THREAD

**Issue by odscjames** [open-contracting/kingfisher-scrape#178]
```
  File "/tmp/kingfisher-1564572422-dyndx_3i.egg/kingfisher_scrapy/spiders/buenos_aires.py", line 40, in parse
    data = zip_file.open('bsas_release.json').read()
  File "/usr/lib/python3.6/zipfile.py", line 1375, in open
    zinfo = self.getinfo(name)
  File "/usr/lib/python3.6/zipfile.py", line 1304, in getinfo
    'There is no item named %r in the archive' % name)
```

```
2019-07-31 15:29:03 (446 KB/s) - ‘2019-7-26-bsas_release_package.zip’ saved [52799528/52799528]

root@process /tmp # unzip 2019-7-26-bsas_release_package.zip 
Archive:  2019-7-26-bsas_release_package.zip
  inflating: 2019-7-26-bsas_release_package.json  
```

**Comment by odscjames** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [buenos_aires crashes because ZIP file is named unexpectedly](https://trello.com/c/XFJRxvGz/54-buenosaires-crashes-because-zip-file-is-named-unexpectedly)

**Comment by odscjames** [comment]
Ok, I've just been looking at the ZIP files in more detail.

A lot of them contain CSV's, not JSON's.

The CSV's at first glance look like they have been generated from some tool we would recognise - here are the headers from one of them:

`ocid,id,parties/0/name,parties/0/id,parties/0/identifier/scheme,parties/0/identifier/id,parties/0/identifier/legalName,parties/0/roles,parties/0/identifier/uri,parties/0/address/streetAddress,parties/0/address/locality,parties/0/address/region,parties/0/address/postalCode,parties/0/address/countryName,parties/0/contactPoint/name,parties/0/contactPoint/email,parties/0/contactPoint/telephone,parties/0/contactPoint/faxNumber,parties/0/contactPoint/url`

However, it terms of writing some kind of automatic scraper here there are problems. 

Tagging @duncandewhurst @mrshll1001 @pindec for discussion.

**Comment by odscjames** [comment]
Also @yolile @romifz

**Comment by jpmckinney** [comment]
Which ZIP files contain CSVs? My understanding of the scraper is that it picks out the resource with `"format": "JSON"`, and there's only one of those, and it contains JSON.

We're in contact with them, and from memory they use Flatten Tool to create the CSVs, such that the JSON and CSVs express the same information.

**Comment by jpmckinney** [comment]
That one JSON file has a file named `2019-7-26-bsas_release_package.json`. I think the logic can just be changed to open any JSON file in the ZIP file, instead of looking for one with a specific name.

**Comment by odscjames** [comment]
The scraper does check for format==JSON, your right, sorry. I was checking all Zip files to try and understand the structure of the API and missed this. In this case, 

> Local-Load command should have option to mark collection store ended after load, optional flag.

Should do it.

**Comment by jpmckinney** [comment]
Closed by #179
